### DISCUSSION TITLE
1.20.2 Neo Or MC Forge?

### DISCUSSION THREAD

**Discussion by SirEndii** [cc-tweaked/CC-Tweaked#1620] [General]
Greetings, 
I have a small question. Will CC:T for 1.20.2 target NeoForge or Minecraft Forge? I want to target Neo for 1.20.2 and I hope CC:T will do that too.

Just asking to know if I can begin the neo port

**Comment by SquidDev** [comment]
NeoForged. There's some preliminary work in #1618, though it's currently pretty non-functional (blocked by https://github.com/neoforged/NeoGradle/issues/12 and https://github.com/neoforged/NeoGradle/issues/13 amongst others).

I'm not expecting much to change API wise - deprecated methods will be removed, and there will be some changes to peripheral registration (`IPeripheralProvider` will be entirely replace by capabilities, due to the changes in https://github.com/neoforged/NeoForge/pull/73).

Hoping to get a release out in a couple of weeks, but will depend on how the work in #1615 goes (and what state NF is by then).

**Reply by SirEdvin** [reply]
I hope priority of custom peripheral provides on top of generic peripheral providers would still work

**Reply by SquidDev** [reply]
Correct. The logic will stay like before, in that we look up via capabilities (or block lookup on Fabric), and fall back to generic peripherals if no capability is present.

**Reply by SirEndii** [reply]
Thanks for the quick answer, nice to hear that CC:T will switch to Neo
